src/content/contribute/writing-a-scaffold.md

```diff
@@ -0,0 +1,400 @@
+      }
+    });
+  }
+};
+```
+
+Sweet! We've now abstracted some part of the code that's probably gonna be really big. We can almost call ourselves engineers now! Let's go ahead and add another question, like asking for an entry point.
+
+```js
+const Generator = require('yeoman-generator');
+const List = require('@webpack-cli/webpack-scaffold').List;
+const Input = require('@webpack-cli/webpack-scaffold').Input;
+const createDevConfig = require('./dev-config');
+
+module.exports = class WebpackGenerator extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
+    opts.env.configuration = {
+      dev: {
+        webpackOptions: {}
+      }
+    };
+  }
+
+  prompting() {
+    return this.prompt([
+      List('confirm', 'Welcome to the demo scaffold! Are you ready?', ['Yes', 'No', 'Pengwings']),
+      Input('entry', 'What is the entry point in your app?')
+    ]).then (answer => {
+      if(answer['confirm'] === 'Pengwings') {
+        this.options.env.configuration.dev.webpackOptions = createDevConfig(answer);
+      }
+    });
+  }
+};
+```
+
+These answers aren't well known on their own for us, so let's go ahead and create our config.
+
+
+## Some more configs
+
+Let's start by looking at `dev-config.js`. We have some answers, now we want to use them to build up an config. How do we do that? Evidently, we should mount our values on the variable we've declared, with some properties we want to build up. We also want to use the answers for the entry prop. I've also taken the liberty to add an output property that has a `filename`.
+
+T> With string values, you need to wrap your strings once again. This is because we can declare some other functionality, using only " ", while " 'Mystring' " resolves to a string.
+
+__dev-config.js__
+
+```js
+module.exports = function createDevConfig(answer) {
+  let entryProp = answer.entry ? ( '\'' + answer.entry + '\'') : '\'index.js\'';
+  let devConfig = {
+    entry: entryProp,
+    output: {
+      filename: '\'[name].js\''
+    }
+  };
+  return devConfig;
+};
+
+```
+
+Try running `webpack init webpack-scaffold-demo`, and now you're seeing your first scaffold!
+
+
+
+## Basic Scaffold
+
+Cool. Now we've got an initial scaffold. Let's add the rest of our options! For the context, let's say we've got a `path.join` we want to make use of. For this, you use a single quote string.
+
+```js
+module.exports = function createDevConfig(answer) {
+  let entryProp = answer.entry ? ( '\'' + answer.entry + '\'') : '\'index.js\'';
+  let devConfig = {
+    entry: entryProp,
+    output: {
+      filename: '\'[name].js\''
+    },
+    context: 'path.join(__dirname, "src")'
+  };
+  return devConfig;
+};
+```
+
+
+
+## Add more functionality
+
+We're ready to add an plugin, as a last piece of the puzzle. For this, I'm going to create an utility for commonsChunk, based on the input from the user. I'll start by adding another question in our prompt.
```

Comment:
an plugin -> a plugin
as a last piece -> as the last piece
an utility -> a utility
in our prompt -> to our prompt